### ISSUE TITLE
Mysql2::Result#count always returns 0 when using stored procedures and MULTI_STATEMENTS

### ISSUE THREAD

**Issue by key88sf** [brianmario/mysql2#406]
Gem versions after 0.3.11 all have this issue.

    db_config = Rails.application.config.database_configuration["development"].symbolize_keys
    client = Mysql2::Client.new(db_config.merge(:flags => Mysql2::Client::MULTI_STATEMENTS, :cache_rows => false))
    sql = "CALL my_proc();" # returns several rows
    result = client.query(sql, :stream => true)
    result.count # 0 (but should be non-zero)

**Comment by sodabrew** [comment]
Ok, I can reproduce this. It appears to be an incompatibility between `:stream => true` and `:flags => Mysql2::Client::MULTI_STATEMENTS`.

**Comment by sodabrew** [comment]
Reading through http://dev.mysql.com/doc/refman/5.0/en/mysql-use-result.html and looking at the code a bit, I am pretty sure we _don't_ correctly handle multi statements streaming correctly. I may need to kick this out by a milestone in order to put some time into it.

**Comment by sodabrew** [comment]
@brianmario Hey, this looks weird. Off the cuff, would you expect that trying to iterate the rest of the results would kill the count?

``` ruby
        @multi_client = Mysql2::Client.new(DatabaseCredentials['root'].merge(:flags => Mysql2::Client::MULTI_STATEMENTS))
        result = @multi_client.query("SELECT 1 AS 'set_1'; SELECT 2 AS 'set_2'", :stream => true, :cache_rows => false)
        result.first.should eql({ 'set_1' => 1 })
        @multi_client.more_results?.should be_true

        result.count.should == 1
        result.each { |r| } # consume the end-of-results packet
        result.count.should == 1
```

Fails with (writing this as part of a new unit test)
```
  1) Mysql2::Client#query Multiple results sets should stream multiple result sets
     Failure/Error: result.count.should == 1
       expected: 1
            got: 0 (using ==)
     # ./spec/mysql2/client_spec.rb:640:in `block (4 levels) in <top (required)>'

```

**Comment by sodabrew** [comment]
Merged #581, will be released in 0.4.0.
